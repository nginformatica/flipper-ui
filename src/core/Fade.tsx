import { Fade as MuiFade } from '@material-ui/core'
import React from 'react'
import { DefaultProps } from './types'

interface FadeProps extends DefaultProps {
    in: boolean
    timeout?: number | { enter?: number, exit?: number }
    children?: React.ReactElement<{}>
}

const Fade = ({
    children,
    padding,
    margin,
    style = {},
    ...otherProps
}: FadeProps) =>
    <MuiFade
        style={ { padding, margin, ...style } }
        { ...otherProps }>
        { children }
    </MuiFade>

export default Fade
