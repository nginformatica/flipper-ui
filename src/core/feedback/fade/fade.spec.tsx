import React from 'react'
import { render, screen } from '@testing-library/react'
import Fade from '.'

describe('Fade', () => {
    it('should render closed', () => {
        render(
            <Fade in={false}>
                <span>Fade Text</span>
            </Fade>
        )
        const element = screen.getByText('Fade Text')

        expect(element).toHaveProperty('style.visibility', 'hidden')
        expect(element).toHaveProperty('style.opacity', '0')
    })

    it('should render opened', () => {
        render(
            <Fade in>
                <span>Fade Text</span>
            </Fade>
        )

        const element = screen.getByText('Fade Text')

        expect(element).not.toHaveProperty('style.visibility', 'hidden')
        expect(element).toHaveProperty('style.opacity', '1')
    })

    it('should match snapshot', () => {
        const { container: opened } = render(
            <Fade in>
                <span>Fade Text</span>
            </Fade>
        )

        const { container: closed } = render(
            <Fade in={false}>
                <span>Fade Text</span>
            </Fade>
        )

        expect(opened).toMatchSnapshot()
        expect(closed).toMatchSnapshot()
    })
})
