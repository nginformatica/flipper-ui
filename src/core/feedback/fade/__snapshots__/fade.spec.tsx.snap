// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fade should match snapshot 1`] = `
<div>
  <div
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <span>
      Fade Text
    </span>
  </div>
</div>
`;

exports[`Fade should match snapshot 2`] = `
<div>
  <div
    style="opacity: 0; visibility: hidden;"
  >
    <span>
      Fade Text
    </span>
  </div>
</div>
`;
